crates/api_models/src/payments.rs

```diff
@@ -1098,51 +1098,100 @@ pub struct GpaySessionTokenData {
 #[serde(rename_all = "lowercase")]
 pub enum SessionToken {
     /// The session response structure for Google Pay
-    Gpay {
-        /// The merchant info
-        merchant_info: GpayMerchantInfo,
-        /// List of the allowed payment meythods
-        allowed_payment_methods: Vec<GpayAllowedPaymentMethods>,
-        /// The transaction info Google Pay requires
-        transaction_info: GpayTransactionInfo,
-    },
+    Gpay(Box<GpayData>),
     /// The session response structure for Klarna
-    Klarna {
-        /// The session token for Klarna
-        session_token: String,
-        /// The identifier for the session
-        session_id: String,
-    },
+    Klarna(Box<KlarnaData>),
     /// The session response structure for PayPal
-    Paypal {
-        /// The session token for PayPal
-        session_token: String,
-    },
+    Paypal(Box<PaypalData>),
     /// The session response structure for Apple Pay
-    Applepay {
-        /// Timestamp at which session is requested
-        epoch_timestamp: u64,
-        /// Timestamp at which session expires
-        expires_at: u64,
-        /// The identifier for the merchant session
-        merchant_session_identifier: String,
-        /// Applepay generates unique ID (UUID) value
-        nonce: String,
-        /// The identifier for the merchant
-        merchant_identifier: String,
-        /// The domain name of the merchant which is registered in Apple Pay
-        domain_name: String,
-        /// The name to be displayed on Apple Pay button
-        display_name: String,
-        /// A string which represents the properties of a payment
-        signature: String,
-        /// The identifier for the operational analytics
-        operational_analytics_identifier: String,
-        /// The number of retries to get the session response
-        retries: u8,
-        /// The identifier for the connector transaction
-        psp_id: String,
-    },
+    Applepay(Box<ApplepayData>),
+}
+
+#[derive(Debug, Clone, serde::Serialize, ToSchema)]
+#[serde(rename_all = "lowercase")]
+pub struct GpayData {
```

Comment:
Change Struct name to `GpaySessionData` or `GpaySessionRequest`

Reply:
As this struct is for SessionTokenResponse, named it as GpaySessionTokenResponse